core/data/src/main/java/com/mifos/core/data/repository/SyncCenterPayloadsRepository.kt

```diff
@@ -9,20 +9,19 @@
  */
 package com.mifos.core.data.repository
 
-import com.mifos.core.entity.center.CenterPayload
-import com.mifos.core.objects.responses.SaveResponse
-import rx.Observable
+import com.mifos.room.entities.center.CenterPayload
+import kotlinx.coroutines.flow.Flow
 
 /**
  * Created by Aditya Gupta on 16/08/23.
  */
 interface SyncCenterPayloadsRepository {
 
-    fun allDatabaseCenterPayload(): Observable<List<CenterPayload>>
+    fun allDatabaseCenterPayload(): Flow<List<CenterPayload>>
 
-    fun createCenter(centerPayload: CenterPayload): Observable<SaveResponse>
+    suspend fun createCenter(centerPayload: CenterPayload?)
 
-    fun deleteAndUpdateCenterPayloads(id: Int): Observable<List<CenterPayload>>
+    fun deleteAndUpdateCenterPayloads(id: Int): Flow<List<CenterPayload>>
 
-    fun updateCenterPayload(centerPayload: CenterPayload): Observable<CenterPayload>
+    suspend fun updateCenterPayload(centerPayload: CenterPayload)
```

Comment:
If you believe a function doesn’t return anything, and you make it suspend while removing the return type, be sure to also remove it from the repository implementation. Additionally, follow it up the chain and do the same where the implementation class calls it.

The unnecessary return type isn’t needed because in Kotlin, functions that don’t return meaningful values implicitly return Unit. Keeping an explicit return where it’s not required just adds confusion and mess without adding value.